### PR TITLE
#792 Make sure any assert main message is displayed into dynawo logs …

### PR DESCRIPTION
**Pull request by rosiereflo**
…and perform algorithm checks only after time step has ended

### CODE REVIEW

**Comment by rosiereflo** [comment]
also solve #793

**Comment by rosiereflo** [comment]
Sonar failure can be ignored as it is about duplication in ModelManager which is independent from this modification

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py [hunk 1]
```diff
@@ -231,9 +231,8 @@ def find_value_in_map(a_map, the_key):
 #
 # @return : the key associated to the value, if the value exists, None otherwise
 def find_key_in_map(a_map, the_value):
-    for key, value in a_map.items():
-        if value == the_value:
-            return key
+    if the_key in a_map:
```
Comment (adriguir) [inline dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py:234]: "the_key" doesn't seem to be defined. Is it normal?

Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py]: Yes, I badly reported it into the script

Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py]: Yes = you are correct this is wrong :) it is solved now.

**Pull request merged**